File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.java
class Solution {
    public int findTheWinner(int n, int k) {
        List<Integer> s = new ArrayList<>();

        // Initialize list
        for (int i = 1; i <= n; i++) {
            s.add(i);
        }

        int idx = 0;

        while (s.size() > 1) {
            idx = (idx + k - 1) % s.size();
            s.remove(idx);
        }

        return s.get(0);
    }
}

File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.cpp
class Solution {
public:
    int findTheWinner(int n, int k) 
    {
        cin.tie(0);cout.tie(0);
        
        set<int> s;int idx=0;

        for (int i = 1; i <= n; i++) s.insert(i);

        while (s.size() > 1) 
        {
            idx = (idx + k - 1) % s.size();
            auto it = s.begin();advance(it,idx);
            it = s.erase(it);
        }
        return *s.begin();
    }
};